base/heartbeat.go

```diff
@@ -159,34 +123,52 @@ func (h *couchbaseHeartBeater) Stop() {
 	}
 }
 
-// Stop sending heartbeats
-func (h *couchbaseHeartBeater) StopSendingHeartbeats() {
-	close(h.heartbeatSendCloser)
+// Send initial heartbeat, and start goroutine to schedule sendHeartbeat invocation
+func (h *couchbaseHeartBeater) startSendingHeartbeats() error {
+	if err := h.sendHeartbeat(); err != nil {
+		return err
+	}
+
+	ticker := time.NewTicker(h.heartbeatSendInterval)
+
+	go func() {
+		defer FatalPanicHandler()
+		defer func() { h.sendActive.Set(false) }()
+		h.sendActive.Set(true)
+		for {
+			select {
+			case _ = <-h.terminator:
+				ticker.Stop()
+				return
+			case <-ticker.C:
+				if err := h.sendHeartbeat(); err != nil {
+				}
```

Comment:
Would it be worth logging a warning here?

Reply:
Makes sense, have added.